File: backend/src/main/java/project/backend/domain/member/dto/event/ProfileUpdateEvent.java
package project.backend.domain.member.dto.event;

import java.util.Optional;
import project.backend.domain.imagefile.ImageFile;
import project.backend.domain.member.entity.Member;

public record ProfileUpdateEvent(
	Long userId,
	String nickname,
	String profileImageUrl
) {

	public static ProfileUpdateEvent of(Member member) {
		return new ProfileUpdateEvent(
			member.getId(),
			member.getNickname(),
			member.getProfileImage()
		);
	}

}

File: backend/src/main/java/project/backend/domain/member/dto/event/ProfileUpdateEvent.kt
package project.backend.domain.member.dto.event

import project.backend.domain.member.entity.Member

data class ProfileUpdateEvent(
    val userId: Long,
    val nickname: String,
    val profileImageUrl: String
) {
    companion object {
        fun of(member: Member): ProfileUpdateEvent {
            return ProfileUpdateEvent(
                member.id!!,
                member.nickname,
                member.profileImage
            )
        }
    }
}